File: eco-core/core-plugin/src/main/java/com/willfp/ecoarmor/sets/meta/ArmorSlot.java
package com.willfp.ecoarmor.sets.meta;

import lombok.Getter;
import org.bukkit.Material;
import org.bukkit.inventory.EquipmentSlot;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public enum ArmorSlot {
    /**
     * Helmet.
     */
    HELMET(EquipmentSlot.HEAD),

    /**
     * Chestplate.
     */
    CHESTPLATE(EquipmentSlot.CHEST),

    /**
     * Elytra.
     */
    ELYTRA(EquipmentSlot.CHEST),

    /**
     * Leggings.
     */
    LEGGINGS(EquipmentSlot.LEGS),

    /**
     * Boots.
     */
    BOOTS(EquipmentSlot.FEET);

    /**
     * The equipment slot.
     */
    @Getter
    private final EquipmentSlot slot;

    ArmorSlot(@NotNull final EquipmentSlot slot) {
        this.slot = slot;
    }

    /**
     * Get ArmorSlot from item.
     *
     * @param itemStack The item.
     * @return The slot, or null.
     */
    @Nullable
    public static ArmorSlot getSlot(@Nullable final ItemStack itemStack) {
        if (itemStack == null) {
            return null;
        }

        Material material = itemStack.getType();
        String[] split = material.name().toLowerCase().split("_");
        String name = split[split.length - 1];

        return switch (name) {
            case "helmet", "head" -> HELMET;
            case "chestplate" -> CHESTPLATE;
            case "elytra" -> ELYTRA;
            case "leggings" -> LEGGINGS;
            case "boots" -> BOOTS;
            default -> null;
        };
    }

    /**
     * Get ArmorSlot from name.
     *
     * @param name The name.
     * @return The slot, or null.
     */
    @Nullable
    public static ArmorSlot getSlot(@NotNull final String name) {
        return switch (name.toLowerCase()) {
            case "helmet" -> HELMET;
            case "chestplate" -> CHESTPLATE;
            case "elytra" -> ELYTRA;
            case "leggings" -> LEGGINGS;
            case "boots" -> BOOTS;
            default -> null;
        };
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/ecoarmor/sets/ArmorSlot.kt
package com.willfp.ecoarmor.sets

import org.bukkit.inventory.EquipmentSlot
import org.bukkit.inventory.ItemStack
import java.util.*

enum class ArmorSlot(
    val slot: EquipmentSlot
) {
    HELMET(EquipmentSlot.HEAD),
    CHESTPLATE(EquipmentSlot.CHEST),
    ELYTRA(EquipmentSlot.CHEST),
    LEGGINGS(EquipmentSlot.LEGS),
    BOOTS(EquipmentSlot.FEET);

    companion object {
        @JvmStatic
        fun getSlot(itemStack: ItemStack?): ArmorSlot? {
            if (itemStack == null) {
                return null
            }
            val material = itemStack.type

            return getSlot(material.name)
        }

        @JvmStatic
        fun getSlot(name: String): ArmorSlot? {
            return when {
                name.contains("HELMET", true) || name.contains("HEAD", true)
                        || name.contains("SKULL", true)
                        || name.contains("PUMPKIN", true) -> HELMET
                name.contains("CHESTPLATE", true) -> CHESTPLATE
                name.contains("ELYTRA", true) -> ELYTRA
                name.contains("LEGGINGS", true) -> LEGGINGS
                name.contains("BOOTS", true) -> BOOTS
                else -> null
            }
        }
    }
}